### DISCUSSION TITLE
ACCESS VIOLATION-solved

### DISCUSSION THREAD

**Discussion by EmreDumlu-UM** [Anura3D/Anura3D_OpenSource#77] [General]
Dear All, 

To begin with, starting to solve the 11th example "Slope failure under transient hydraulic loads" by double-point formulation, I got two different errors as follows:

1. Since the example has a slope material and water, firstly. I gave the slope 3 Solid MP and 3 Liquid MP.  For the liquid (reservoir), 0 solid MP and 3 liquid MP.  Then I assigned the slope material to the dam and the Liquid material to Reservoir. After clicking the calculation.bat I got the "Acces violation error" in Figure 1.

![Acces violation](https://user-images.githubusercontent.com/125078884/223807357-131750ef-30dd-4a16-9a58-ac473353ec01.png)

The related figures are given below, as well:

![material_slope](https://user-images.githubusercontent.com/125078884/223807420-69444c73-cde7-4fa2-b2e5-8fc684ced6b0.png)

![material point specifications](https://user-images.githubusercontent.com/125078884/223807442-40bda432-5efe-4146-a586-8565e0dacd82.png)

![calculationdata_1](https://user-images.githubusercontent.com/125078884/223807490-f068aad0-1ec3-4308-a4ab-173d7780796c.png)

![calculationdata_2](https://user-images.githubusercontent.com/125078884/223807514-31431551-23ad-43d0-bfce-ae0fc309a986.png)

To be honest, I am stuck and so far I have not gotten through the issue. Is there anyone who encountered this issue before?

Best regards.

**Comment by EmreDumlu-UM** [comment]
2.  If I describe the water with "the slope material" as 0 Solid Points and 3 Liquid Points the error says "Material point specification of element 471 does not coincide with material type". The related figure is given below:

![Material point specification of element 471 does not coincide with material type](https://user-images.githubusercontent.com/125078884/223809577-f52dcc8b-ecc5-450e-9a11-228cdbb9b658.png)

![element 471](https://user-images.githubusercontent.com/125078884/223811997-9ef522b3-118e-4d49-bc74-effea61ddf5d.png)

The element is right at the corner at the intersection between the water and the dam's surface.

Is there anyone who encountered this type of error before?

Best regards.

**Comment by murphy-john** [comment]
It looks like you're trying to use the double-point formulation. You could try to do two things:
1) Make your model single-point with a liquid boundary condition and instead of a liquid material. 
2) If you want to stay with the double-point, you'll need to specify two materials, one for the slope and one for the water. You'll change the material type of the water to "Liquid" and keep the material points specified in the water as 0 SMP and 3 LMP.

**Reply by EmreDumlu-UM** [reply]
Dear Murphy,

Thank you for your suggestion. I have already archived the results by following the sample in single-point. It does match.

When applying your second suggestion, it gives me an access violation error shown in Figure 1.  

Best regards,

**Reply by murphy-john** [reply]
Ah my mistake, this can happen when a file is locked by Windows or it can't be located in the directory. The best way to find the file Fortran can't locate is by [debugging the code](https://learn.microsoft.com/en-us/visualstudio/debugger/debugging-absolute-beginners?view=vs-2022&tabs=csharp).

**Reply by Francisliii** [reply]
Hi,

Have you figured out a solution to the "Program Exception - access violation" issue? I had the same error whenever I applied uniform solid traction to 3D models.

Thanks,
Francis

**Reply by EmreDumlu-UM** [reply]
Hi Francis,

I solved. It caused the double-point formulation. I had a mistake while defining the liquid material with zero solid MP. Then I defined the water as a liquid and changed the calculation type MPM- Material Point Integration. It worked!

Thank you for your interest.

**Reply by Francisliii** [reply]
Hi, thanks for your reply.

Francis

**Reply by murphy-john** [reply]
Hello Emre-ncche, yes for the double-point you must use MPM-MP integration. The MPM-MIXED has not been implemented.
